### PR TITLE
chore(test): Minor tweaks to migrated WebdriverIO tests

### PR DESCRIPTION
**Pull request by christian-bromann**
## What is the current behavior?
I saw some weird behavior when reviewing #5478. We moved forward and merged the PR achieving the same assertions in a different way, however I went back and fixed some things in WebdriverIO to get it working.

## What is the new behavior?
The way @alicewriteswrongs original expected WebdriverIO to work, works now. I updated the tests to reflect this. I also fixed some linting issues (mostly import orders).

## Documentation

n/a

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Testing

n/a

## Other information

n/a

### CODE REVIEW

File: test/wdio/lifecycle-basic/cmp-a.test.tsx [hunk 1]
```diff
@@ -11,37 +11,32 @@ describe('lifecycle-basic', function () {
...
     await expect(loads[1]).toHaveText('componentWillLoad-b');
     await expect(loads[2]).toHaveText('componentWillLoad-c');
     await expect(loads[3]).toHaveText('componentDidLoad-c');
     await expect(loads[4]).toHaveText('componentDidLoad-b');
     await expect(loads[5]).toHaveText('componentDidLoad-a');
 
-    let updates = await $('.lifecycle-updates-a').$$('li');
-    await expect(updates).not.toBeExisting();
+    let updates = $('.lifecycle-updates-a').$$('li');
+    await expect(loads).toBeElementsArrayOfSize(6);
```
Comment (rwaskiewicz) [inline test/wdio/lifecycle-basic/cmp-a.test.tsx:24]: Why do we change the assertion here to be against the field in `loads` instead of `updates` here?

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]
Can we update the commit msg + PR title to be `chore(test):`? While this fixes some things for our tests internally, it won't have an affect on end users & shouldn't show up in the autogenerated changelog

**Review by alicewriteswrongs — APPROVED** [review]
looks good, other than @rwaskiewicz's comment

File: test/wdio/lifecycle-basic/cmp-a.test.tsx (hunk 1 above)
Comment (christian-bromann) [reply test/wdio/lifecycle-basic/cmp-a.test.tsx]: Oh, my bad! It is actually suppose to be `await expect(updates).toBeElementsArrayOfSize(0);`

File: test/wdio/package.json [hunk 2]
```diff
@@ -11,9 +11,9 @@
     "@stencil/core": "file:../..",
     "@stencil/sass": "^3.0.10",
     "@types/node": "^16.11.7",
-    "@wdio/browser-runner": "^8.33.1",
-    "@wdio/cli": "^8.33.1",
-    "@wdio/globals": "^8.33.1",
+    "@wdio/browser-runner": "^8.34.1",
+    "@wdio/cli": "^8.34.1",
+    "@wdio/globals": "^8.34.1",
```
[the remark this answers is not in the document]
Comment (christian-bromann) [reply test/wdio/package.json:16]: All `@wdio/*` packages as well as `expect-webdriverio`

**Comment by christian-bromann** [comment]
Blocked on https://github.com/webdriverio/webdriverio/issues/12528

**Review by christian-bromann — COMMENTED** [review]
Some comments on particular patterns I picked up:

## wrapping into `browser.waitUntil`

if certain sequences are uncertain due to their async nature, it might make sense to wrap them in a `waitUntil` block which re-executes the function until the condition is met. It can be used to use `querySelector` and attempt to fetch an element until found, e.g.:

```ts
const elem: HTMLElement = await browser.waitUntil(async () => {
  return document.body.querySelector('form');
});
```

Here if `document.body.querySelector('form')` doesn't find an element in the first attempt, it retries until it gets there. Consequently if the assertion fails for some reason the block will be retried as well. If you don't return a value, make sure to return a truthy value at the end so the condition is noted as successful.

## Querying/Asserting Children

We have some tests where we would do the following:

```ts
    const host = document.querySelector('import-aliasing');

    expect(host.children[0].textContent).toBe('My name is John');
    expect(host.children[1].textContent).toBe('Name changed 0 time(s)');
```
@rwaskiewicz found a practical way of getting ahold of the children using WebdriverIO primitives by using a `> *` at the end of the query, e.g.:

```ts
    const children = $$('import-aliasing > *');
    await expect(children[0]).toHaveText('My name is John');
    await expect(children[1]).toHaveText('Name changed 0 time(s)');
```
However I will add a `children()` command to WebdriverIO so we can just do:

```ts
    const children = $('import-aliasing').children();
    await expect(children[0]).toHaveText('My name is John');
    await expect(children[1]).toHaveText('Name changed 0 time(s)');
```

__Note:__ I've never considered this really a usecase as WebdriverIO was primarily used for e2e testing where you usually would query an element straight rather than going through the DOM tree. With component and unit testing in the browser this becomes more crucial.

## Retries

I think this is one of the key points that makes the test more stable. While one could say, retries are an antipattern, in our case, it is cheap to do and has a big impact. I really can't say in some cases why certain tests are unreliable. A lot is executing very fast which I believe can also impact the browser rendering or other components we don't have control over.

File: test/wdio/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (rwaskiewicz) [reply test/wdio/package.json]: I've got a moment - I can take care of this

**Pull request merged**